src/app/src/main/kotlin/org/vaccineimpact/api/app/app_start/Router.kt

```diff
@@ -67,14 +69,21 @@ class Router(val config: RouteConfig,
     private fun getWrappedRoute(endpoint: EndpointDefinition): Route
     {
         return Route({ req, res ->
-            repositoryFactory.inTransaction { repos ->
-                invokeControllerAction(endpoint, DirectActionContext(req, res), repos)
+            try
+            {
+                repositoryFactory.inTransaction { repos ->
+                    invokeControllerAction(endpoint, DirectActionContext(req, res), repos)
+                }
+            }
+            finally
+            {
+                req.consumeRemainder()
```

Comment:
Currently we have this `try...catch` in three places, but when we finish switching over to the new style controllers that will drop down to 1.